Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorPickEntitySelection.cpp

```diff
@@ -14,17 +14,18 @@
 
 namespace AzToolsFramework
 {
+    AZ_CLASS_ALLOCATOR_IMPL(EditorPickEntitySelectionHelper, AZ::SystemAllocator, 0)
     AZ_CLASS_ALLOCATOR_IMPL(EditorPickEntitySelection, AZ::SystemAllocator, 0)
 
-    EditorPickEntitySelection::EditorPickEntitySelection(
+    EditorPickEntitySelectionHelper::EditorPickEntitySelectionHelper(
```

Comment:
thought: I wonder if these changes will actually make this type easier to test too which is really nice (not something you have to do now but it'd be great in future to have some sanity checks this code is doing what we expect)